pre-sync/oci-image-verification/README.md

```diff
@@ -0,0 +1,158 @@
+
+### Cosign keys
+
+Generate cosign.key and cosign.pub pairs:
+
+```shell
+cosign generate-key-pair
+```
+
+Create on cluster secret:
+
+```shell
+kubectl create secret generic cosign-key --from-file=cosign.pub -n signature-verification
+```
+
+### Signature Verification Server Authentication
+
+Generate tls.crt and tls.key:
+
+```shell
+openssl req -nodes -x509 -sha256 -newkey rsa:4096 \
+-keyout tls.key \
+-out tls.crt \
+-days 356 \
+-subj "/CN=signature-verification-service.signature-verification.svc"  \
+-addext "subjectAltName = DNS:signature-verification-service,DNS:signature-verification-service.signature-verification.svc,DNS:signature-verification-service.signature-verification"
+```
+
+Create on cluster secret:
+
+```shell
+kubectl create secret tls webhook-tls --cert=tls.crt --key=tls.key -n signature-verification
+```
+
+## Deploy the signature verification server and ValidatingWebhookConfiguration
+
+#### In signature-verification-deployment.yaml file
+
+Replace the <PROJECT_ID> with the project name where signature verification server image is hosted.
+
+Replace the <SIGNATURE_VERIFICATION_SERVER_IMAGE_URL> with the URL of the signature verification server image.
+
+#### In signature-verification-validatingwebhookconfiguration.yaml file
+
+Replace <CA_BUNDLE> with the base64-encoded content of tls.crt: `cat tls.crt | base64 -w 0`.
+
+#### Apply the manifests to cluster.
+
+```shell
+kubectl apply -f signature-verification-deployment.yaml -n signature-verification
+kubectl apply -f signature-verification-validatingwebhookconfiguration.yaml
+```
+
+## Test the image signature verification
+
+- [Install Config Sync], configure it to [sync from an unsigned OCI image].
+
+- Look for errors in signature verification server log `kubectl logs deployment signature-verification-server -n signature-verification`:
+
+```text
+main.go:69: error during command execution: no signatures found
+```
+
+- Config Sync will additionally report an `APIServerError` when `nomos status` is run.
+
+```text
+Error:   KNV2002: failed to patch RootSync annotations: admission webhook "imageverification.webhook.com" denied the request: Image verification failed: image verification failed for <SOURCE_IMAGE_URL>: no signatures found
+Patch content: {"metadata":{"annotations":{"configsync.gke.io/image-to-sync":"<SOURCE_IMAGE_URL>"}}}
+```
+
+- Sign the same source image using Cosign
+
+```shell
+cosign sign <SOURCE_IMAGE_URL> --key cosign.key
+```
+
+- Once the image is correctly signed, the signature verification server should successfully verify it. This will result in:
+
+  - The Config Sync source error being cleared.
+  - The `configsync.gke.io/image-to-sync` annotation on the RootSync object being updated to reflect the new signed image.
+
+- You can verify this by inspecting the RootSync object
+```shell
+kubectl get rootsync <ROOT_SYNC_NAME> -n config-management-system -oyaml
+```
+Or the RepoSync object
```

Comment:
Slight formatting issue. This or will in different indentation as the list item. Maybe just combine them together as: 

- You can verify this by inspecting the RootSync object or the RepoSync object
```shell
kubectl get rootsync <ROOT_SYNC_NAME> -n config-management-system -oyaml
kubectl get reposync <REPO_SYNC_NAME> -n <REPO_SYNC_NAMESPACE> -oyaml
```

Reply:
Done